stardis/io/model/util.py

```diff
@@ -56,3 +58,29 @@ def create_scaled_solar_profile(
 
     # Return scaled mass fractions by dividing by total mass. Implicitly lowers the hydrogen abundance so that the total mass fraction is 1.
     return solar_values.div(solar_values.sum(axis=0))
+
+
+def rescale_chemical_mass_fractions(composition, elements, scale_factors):
+    """
+    Renormalizes the composition after multiplying the specified elements by a list of scale factors.
+
+    Args:
+    composition: The composition object to rescale.
+        nuclides: The nuclides to rescale by specified scale factors.
+        scale_factors: How much to rescale the specified nuclides by before renormalizing.
+    """
+
+    new_mass_fractions = composition.nuclide_mass_fraction.copy().T
+
+    for element, scale_factor in zip(elements, scale_factors):
+        if not isinstance(element, int):
+            element = element_symbol2atomic_number(element)
+        logging.info(
+            f"Rescaling {atomic_number2element_symbol(element)} by {scale_factor}"
+        )
```

Comment:
Perhaps there should be warning here just in case that the user provide a element number or symbol that is not in the composition.